### DISCUSSION TITLE
root certificate path

### DISCUSSION THREAD

**Discussion by zhaoyanliang8** [justcoding121/Titanium-Web-Proxy#894] [General]
`Assembly.GetEntryAssembly().Location` returns an empty string in a single file application.

https://github.com/justcoding121/titanium-web-proxy/blob/2825191dc8734108072e5f6af98b4a9ce3554492/src/Titanium.Web.Proxy/Certificates/Cache/DefaultCertificateDiskCache.cs#L129
https://docs.microsoft.com/en-us/dotnet/core/deploying/single-file

**Comment by justcoding121** [comment]
Good catch ? that was a real single-file publish gotcha.

It's been fixed: if `Assembly.GetEntryAssembly()?.Location` is empty we fall back to `AppContext.BaseDirectory`.

You can also pin the path yourself:

```csharp
proxyServer.CertificateManager.PfxFilePath = Path.Combine(AppContext.BaseDirectory, "rootCert.pfx");
```

Closing as resolved.
